src/Columns/IColumn.h

```diff
@@ -367,6 +376,25 @@ class IColumn : public COW<IColumn>
         throw Exception("Method structureEquals is not supported for " + getName(), ErrorCodes::NOT_IMPLEMENTED);
     }
 
+    static constexpr auto DEFAULT_ROWS_SEARCH_STEP = 8;
+    static constexpr auto MIN_ROWS_TO_SEARCH_DEFAULTS = DEFAULT_ROWS_SEARCH_STEP * 16;
+    static constexpr auto DEFAULT_RATIO_FOR_SPARSE_SERIALIZATION = 0.95;
+
+    /// Returns number of values in column, that equal to default value of column.
+    /// Checks every @step-th value. So, if step is not 1, returns number,
+    /// that lower than actual. 0 means, that such statistic is unknown for column.
+    virtual size_t getNumberOfDefaultRows(size_t /* step */) const { return 0; }
```

Comment:
Equal-distance sampling might have weird boundary effects. Maybe we need to add some more complex sampling, but I can't really comment on this. I know blue noise might be helpful when sampling a signal, but not sure how applicable it is here (see e.g. https://bartwronski.com/2016/10/30/dithering-part-two-golden-ratio-sequence-blue-noise-and-highpass-and-remap/).

This interface looks a little bit too specific for equal distance sampling, maybe make it return a float with percentage, and take a float with which percentage of column to sample? I think it would also convey the purpose more clearly.